scanpy/preprocessing/_normalization.py

```diff
@@ -27,6 +32,217 @@ def _normalize_data(X, counts, after=None, copy=False):
+
+    # clip
+    residuals = np.clip(residuals, a_min=-clip, a_max=clip)
+
+    return residuals
+
+
+def normalize_pearson_residuals(
+    adata: AnnData,
+    theta: float = 100,
+    clip: Union[Literal['auto', 'none'], float] = 'auto',
+    layer: Optional[str] = None,
+    copy: bool = False,
+    inplace: bool = True,
+) -> Optional[Dict[str, np.ndarray]]:
+    """\
+    Computes analytic Pearson residuals, assuming a negative binomial offset
+    model with overdispersion theta shared across genes. By default, residuals
+    are clipped to sqrt(n) and overdispersion theta=100 is used.
+
+    Params
+    ------
+    adata
+        The annotated data matrix of shape `n_obs` × `n_vars`.
+        Rows correspond to cells and columns to genes.
+    theta
+        The NB overdispersion parameter theta. Higher values correspond to
+        less overdispersion (var = mean + mean^2/theta), and `theta=np.Inf`
+        corresponds to a Poisson model.
+    clip
+        Determines if and how residuals are clipped:
+        
+        * If `'auto'`, residuals are clipped to the interval
+        [-sqrt(n), sqrt(n)], where n is the number of cells in the dataset
+        (default behavior).
+        * If any scalar c, residuals are clipped to the interval [-c, c]. Set
+        `clip=np.Inf` for no clipping.
+        
+    layer
+        Layer to normalize instead of `X`. If `None`, `X` is normalized.
+    inplace
+        Whether to update `adata` or return dictionary with normalized copies
+        of `adata.X` and `adata.layers`.
+    copy
+        Whether to modify copied input object. Not compatible with
+        `inplace=False`.
+
+    Returns
+    -------
+    Returns dictionary with Pearson residuals and settings
+    or updates `adata` with normalized version of the original
+    `adata.X` and `adata.layers`, depending on `inplace`.
+
+    """
+
+    if copy:
+        if not inplace:
+            raise ValueError("`copy=True` cannot be used with `inplace=False`.")
+        adata = adata.copy()
+
+    view_to_actual(adata)
+    X = _get_obs_rep(adata, layer=layer)
+    computed_on = layer if layer else 'adata.X'
+
+    msg = 'computing analytic Pearson residuals on %s' % computed_on
+    start = logg.info(msg)
+
+    residuals = _pearson_residuals(X, theta, clip, copy=~inplace)
+    settings_dict = dict(theta=theta, clip=clip, computed_on=computed_on)
+
+    if inplace:
+        _set_obs_rep(adata, residuals, layer=layer)
+        adata.uns['pearson_residuals_normalization'] = settings_dict
+    else:
+        results_dict = dict(X=residuals, **settings_dict)
+
+    logg.info('    finished ({time_passed})', time=start)
+
+    if copy:
+        return adata
+    elif not inplace:
+        return results_dict
+
+
+def normalize_pearson_residuals_pca(
+    adata: AnnData,
+    theta: float = 100,
+    clip: Union[Literal['auto', 'none'], float] = 'auto',
+    n_comps_pca: Optional[int] = 50,
+    random_state_pca: Optional[float] = 0,
```

Comment:
I think the svd solver is also an arg I would expose to user (for reproducibility purposes). In general there could be other, what about using `**kwargs` or `**pca_kwargs` if you have others?

Reply:
actually potentially both `kwargs` could be useful (also for the pearson residuals, e.g. `batch_key` and others).

Reply:
good idea with with the `kwargs_pca`, I added that (here and also to the `recipe`)!

about the `kwargs` for `normalize_pearson_residuals()`: I understood that you suggest to add another argument `kwargs` that allows to further customize the call to `normalize_pearson_residuals()` within the bundle/recipe. 

I would rather not add that here, because I assume that users could customize this easily enough for themselves if they wanted to use `highly_variable_genes(falvor='pearson_residuals')` and `normalize_pearson_residuals()` with different `clip`, `theta`, or `check_values`, and it creates some clutter in the code/docs. 

What do you think?

Reply:
looks good!